File: Data Structures & Algorithms/two-integer-sum/submission-0.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut twoSumMap: HashMap<i32, i32> = HashMap::new();
        for (i, &num) in nums.iter().enumerate(){
            let complement = target - num;
            if let Some(&j) = twoSumMap.get(&complement){
                return vec![j , i as i32];
            }
            twoSumMap.insert(num, i as i32);
        }
        return vec![];
    }
}

File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majorityEl;
        int count=0;
        majorityEl = nums[0];
        for(int i =0; i<nums.size(); i++){
            if(nums[i] == majorityEl){
                count++;
            }else{
                count --;
                if(count<0){
                    majorityEl = nums[i];
                    count = 1;
                }
            }
        }
        return majorityEl;
    }
};